pkg/config/config_test.go

```diff
@@ -15,19 +15,119 @@ import (
+
+	gt := NewGomegaWithT(t)
+
+	signingIdentity, err := NewSigningIdentity(publicKey, privateKey, "test-msp")
+	gt.Expect(err).NotTo(HaveOccurred())
+	configSignature, err := SignConfigUpdate(&common.ConfigUpdate{}, signingIdentity)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	expectedCreator, err := signingIdentity.Serialize()
+	gt.Expect(err).NotTo(HaveOccurred())
+	signatureHeader := &common.SignatureHeader{}
+	err = proto.Unmarshal(configSignature.SignatureHeader, signatureHeader)
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(signatureHeader.Creator).To(Equal(expectedCreator))
+}
+
+func TestCreateSignedConfigupdateEnvelope(t *testing.T) {
+	t.Parallel()
+	gt := NewGomegaWithT(t)
+
+	publicKey, privateKey := generatePublicAndPrivateKey()
+
+	configUpdate := &common.ConfigUpdate{
+		ChannelId: "testchannel",
+	}
+
+	// create signingIdentity
+	signingIdentity, err := NewSigningIdentity(publicKey, privateKey, "test-msp")
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	// create detached config signature
+	configSignature, err := SignConfigUpdate(configUpdate, signingIdentity)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	// create signed config envelope
+	signedEnv, err := CreateSignedConfigupdateEnvelope(configUpdate, signingIdentity, configSignature)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	payload := &common.Payload{}
+	err = proto.Unmarshal(signedEnv.Payload, payload)
+	gt.Expect(err).NotTo(HaveOccurred())
+	// check header channel ID equal
+	channelHeader := &common.ChannelHeader{}
+	err = proto.Unmarshal(payload.GetHeader().GetChannelHeader(), channelHeader)
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(channelHeader.ChannelId).To(Equal(configUpdate.ChannelId))
+	// check config update envelope signatures are equal
+	configEnv := &common.ConfigUpdateEnvelope{}
+	err = proto.Unmarshal(payload.Data, configEnv)
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(len(configEnv.Signatures)).To(Equal(1))
+	expectedSignatures := configEnv.Signatures[0]
+	gt.Expect(expectedSignatures.SignatureHeader).To(Equal(configSignature.SignatureHeader))
+	gt.Expect(expectedSignatures.Signature).To(Equal(configSignature.Signature))
+}
+
+func TestCreateSignedConfigupdateEnvelopeFailures(t *testing.T) {
+	t.Parallel()
+	gt := NewGomegaWithT(t)
+
+	publicKey, privateKey := generatePublicAndPrivateKey()
+
+	configUpdate := &common.ConfigUpdate{
+		ChannelId: "testchannel",
+	}
+
+	// create signingIdentity
+	signingIdentity, err := NewSigningIdentity(publicKey, privateKey, "test-msp")
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	// create detached config signature
+	configSignature, err := SignConfigUpdate(configUpdate, signingIdentity)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	tests := []struct {
+		spec            string
+		configUpdate    *common.ConfigUpdate
+		signingIdentity *SigningIdentity
+		configSignature []*common.ConfigSignature
+		expectedErr     string
+	}{
+		{"when the signing identity is not specified",
```

Comment:
Nit: new line here and in the subsequent test cases.

Reply:
done